src/UIKit/UIAppearance.cs

```diff
@@ -31,7 +31,7 @@ public override bool Equals (object other)
 
 		public override int GetHashCode ()
 		{
-			return ((IntPtr) Handle).ToInt32 ();
+			return HashCode.Combine (Handle);
```

Comment:
This could do like the other methods hashing on the Handle property:

```suggestion
			return Handle.GetHashCode ();
```